(Add) Import 30 blogs (#151)
openebs/website: website/src/blogs/2018-10-22-running-openebs-on-custom-rancher-cluster.md

---
title: Running OpenEBS On Custom Rancher Cluster
slug: running-openebs-on-custom-rancher-cluster
author: Chandan Sagar Pradhan
date: 22-10-2018
tags: Cas, Cloud Native Storage, Kubernetes, Openebs, Rancher
excerpt: In this blog, I will be explaining how to deploy an application in a custom Rancher cluster on an OpenEBS volume.
---

> In this blog, I will be explaining how to deploy an application in a custom Rancher cluster on an OpenEBS volume. **OpenEBS** is a leading open-source storage platform that provides persistent and containerized block storage for DevOps and container environments. **Rancher **is enterprise management for Kubernetes. Before you begin, please make sure all the prerequisites are met.

Reviewer: ```suggestion
In this blog, I will be explaining how to deploy an application in a custom Rancher cluster on an OpenEBS volume. **OpenEBS** is a leading open-source storage platform that provides persistent and containerized block storage for DevOps and container environments. **Rancher** is enterprise management for Kubernetes. Before you begin, please make sure all the prerequisites are met.
```

## Node Requirements:

- Ubuntu 16.04(64-bit)
- 4 vCPUS and 16 GB RAM
- Docker 17.03.2

## Prerequisites:

- Rancher 2.0 UI installed and running.

After the installation of the docker, I used the below command to install Rancher. For more information on installing Rancher visit Rancher docs at [https://rancher.com/docs/rancher/v2.x/en/](https://rancher.com/docs/rancher/v2.x/en/)

    sudo docker run -d --restart=unless-stopped -p 8080:8080 rancher/server:stable

- Minimum of 1 master and 3 worker
- (Optional) 6 nodes for the cluster ( 3 master and 3 workers)
- *iscsiadm* should be present only on kubelet (And verification steps)

### Remove iscsiadm from nodes:

Check the below commands on all worker nodes.

    iscsiadm -V
    docker exec kubelet iscsiadm -V

Sample Output:

    root@worker1 ~ # iscsiadm -V
    iscsiadm version 2.0–873
    root@worker1 ~ # sudo docker exec kubelet iscsiadm -V
    iscsiadm version 2.0–874

> Ifyour output is similar to the sample above, then you have to remove iscsi from the node. OpenEBS target will use the iscsi inside the kubelet. Run the commands below to remove iscsi from the node.

Reviewer 2: ```suggestion
If your output is similar to the sample above, then you have to remove iscsi from the node. OpenEBS target will use the iscsi inside the kubelet. Run the commands below to remove iscsi from the node.
```

    service iscsid stop
    sudo apt remove open-iscsi

### Load iscsi_tcp module:

The above step may remove the iscsi_tcp probe, and after a reboot, the node will not start the iscsi_tcp service, and OpenEBS volume mount will fail. It should be the same with the command below.

    lsmod | grep iscsi

SampleOutput:

    root@worker113:~# lsmod | grep iscsi
    iscsi_tcp 20480 0
    libiscsi_tcp 24576 1 iscsi_tcp
    libiscsi 53248 2 libiscsi_tcp,iscsi_tcp
    scsi_transport_iscsi 98304 2 libiscsi,iscsi_tcp

If your output is similar to the sample above, then, you are good to go. If your output doesn’t have *iscsi_tcp*, you need to follow the below steps to load the *iscsi_tcp* module.

    modprobe iscsi_tcp

You can verify the same from the command below. Now the output should be similar to the sample output mentioned above

    lsmod | grep iscsi

### Persist iscsi_tcp module to load after reboot:

> You can make the kernel load iscsi_tcp automatically every time the node reboots by appending the line iscsi_tcp in /etc/ modules.

Reviewer: ```suggestion
You can make the kernel load iscsi_tcp automatically every time the node reboots by appending the line `iscsi_tcp` in `/etc/` modules.
```

Example:

    # /etc/modules: kernel modules to load at boot time.
    #
    # This file contains the names of kernel modules that should be loaded
    # at boot time, one per line. Lines beginning with “#” are ignored.
    iscsi_tcp

Now if all prerequisites have been met, go ahead with setting up the cluster. You can go directly to deploy the OpenEBS section if you already have a k8s cluster.

### Creating a Custom k8s cluster on Rancher:

- Once you have deployed Rancher, you should be able to access the UI. Login to your rancher UI using your credentials

> ![https://cdn-images-1.medium.com/max/800/0*9hGJnGOsGQCTQHOT](https://lh5.googleusercontent.com/GPlkMiXFCbx2NDcEpBnPts6RyogvKkba2FB-KXK8MJGPNSIUPtc8m0XDgfyEHrFd4gbt0m_pnZlrjJxJyZMpt3n8TG6HleaK-PXx5w2sK79acOpDEG75qx1_ADiqr-IYKiptoWWnEDjmVZJvzg)

Reviewer: please correct the value of image tag `![alt](image-path)`

- Now click on the global tab and then click on add cluster button. We should see the ‘add cluster’ window.

Example:

> ![https://cdn-images-1.medium.com/max/800/0*wvgcOh-f-sej1z7K](https://lh4.googleusercontent.com/8I-bx0annuUna6yqZW4WBKx7B2yJ1HZqAULkMU8Cy0ml4qrDOg8uDdlgvBllT3TrLDLjguVzih3NWm21wa1LEryF-DJJ4C_xMYbrmLvTmvQCvbP38xWNgEGTiR_hHuD78ja3avF1eqh3EHDFkQ)

Reviewer: here also

- Under the add cluster option, click on the custom. Then give the cluster a name. You can customize the cluster under Cluster Options. You can choose Kubernetes version, Network provider and other options. Here, I have selected Kubernetes Version as V.1.11.2-rancher1–1, Network provider as Canal and Cloud Provider as none.

Example:

> ![https://cdn-images-1.medium.com/max/800/0*TUdHPrwikhS2ayyc](https://lh3.googleusercontent.com/HdRA1u7VmMHKX0x3zLGMLCenORq8wyJaOCMo7QROGW77Id8m9gOTLB_Pyj444jYmcE9-MOsyTrt7kDtqx8hsiTVcM63bAb_qGZpOZnOgUSLM6WcvSOxy1FXN2X8DNbfFjZNhno9dLPfJpZwLEw)![https://cdn-images-1.medium.com/max/800/0*WmuNv-q8y2DWCrtf](https://lh4.googleusercontent.com/iFYGuhO9hdQhd8yx3nnznxRIsLQf6jtrcDgyPX6XayuI0JTC0zM4tsCzVrN6HEqZys4qKYHW1rMUiFnoEvPUlFO_ikh2dCKRWt6yjpvVShC8xE7enhy0bpyeNSG1y_KhcNATX70YGdDUulqCZg)

Reviewer: here also, most of the place in this blog.

- Now, click on the Next button; it will open another page. On this page, you can select node roles like etcd/ Control Plane / Worker. Click on `Show advanced options.` Now add the i/p address of the node in the Internal Address section.

Example:
![https://cdn-images-1.medium.com/max/800/0vdEKTs_03NjU_ABz](https://lh5.googleusercontent.com/PxmfXKEX56zJs9wZtN2SE0ssvtVvU74FU12HPhuUu4f0dT1Ctr3pDRe_VBQGsFVCnJWE54CMa3Qp9kDaPa1oXggWyaza90-7HAkwWSyEqm4aCeZZwPths6zNSeBX2TqCyJNA5MsT9SjqCbdkSw)
- Copy the command mentioned in the page to the node, which will be added to the cluster. Once you run the command, the page will show `1 new node registered.`

> ![https://cdn-images-1.medium.com/max/800/0RWPLUNaKkqjytKS2](https://lh3.googleusercontent.com/Nwmz-UPUTa_BGznUiBU_E3KmMNU-1yvh9f1Y3j5M-3hHssizRSMtCNlejgB_qDmVxUeoLDuXqT9EZDtbIuHuCq-7KP66f8scl6f2xE_H_dzd1oFbmEyoX-SUiBFBcjzRF2nhmyQQrd_9Jdx7-w)

Reviewer 2: Please correct the alt texts here
